### PR TITLE
(MODULES-5899) Fix for rubocop violations

### PR DESCRIPTION
**Pull request by glennsarti**
This commit updates the non-test code to repair rubocop violations.  Note that
a rubocop_todo file has been introduced.  These violations should be fixed in
the future but will require more effort than can be expected in this commit.
Later commits will fix the violations and remove the todo entries.

---

Previously the ACL module monkey patched the Win32-Security and Puppet code in
older versions.  Now that the module only supports Puppet 4.7 or above, which
means Win32-Security 0.2.5 or above, the monkey patches are no longer applicable.
This commit removes the monkey patching.

### CODE REVIEW

**Comment by glennsarti** [comment]
Requires an adhoc run before merge.

**Comment by glennsarti** [comment]
Passed adhoc

File: lib/puppet/provider/acl/windows/base.rb [hunk 1]
```diff
@@ -239,11 +236,11 @@ def are_permissions_insync?(current_permissions, specified_permissions, purge_va
...
-          if current_permissions.nil?
-            current_local_permissions = []
-          else
-            current_local_permissions = current_permissions.select { |p| !p.is_inherited? }
-          end
+          current_local_permissions = if current_permissions.nil?
+                                        []
+                                      else
+                                        current_permissions.reject { |p| p.inherited? }
+                                      end
```
Comment (ThoughtCrhyme) [inline lib/puppet/provider/acl/windows/base.rb]: all this whitespace seems odd to me, but if we're cool with it...

Comment (glennsarti) [reply lib/puppet/provider/acl/windows/base.rb]: Yeah... it wants to align the else and end with the if.

**Pull request merged**